pkg/operation/botanist/dnsresources.go

```diff
@@ -16,23 +16,21 @@ package botanist
 
 import (
 	"context"
-
-	gardencorev1beta1helper "github.com/gardener/gardener/pkg/apis/core/v1beta1/helper"
 )
 
 // DeployOwnerDNSResources deploys or deletes the owner DNSRecord resource depending on whether
 // the 'ownerChecks' setting is enabled.
 // * If the ownerChecks is enabled, the DNSRecord resource is deployed (or restored).
 // * Otherwise, it is deleted.
+// TODO(plkokanov): This can be removed in a future version, when all owner
+// DNSRecords have been cleaned up from existing Shoots.
 func (b *Botanist) DeployOwnerDNSResources(ctx context.Context) error {
```

Comment:
Please remove this function along with `MigrateOwnerDNSResources` and simply call `DestroyOwnerDNSRecord` directly.